File: Leetcode/C++/392. Is Subsequence/solution.cpp
// https://leetcode.com/problems/is-subsequence/

#include <string>

using namespace std;

class Solution
{
public:
    bool isSubsequence(string s, string t)
    {
        int count = 0;
        for (auto char_t : t)
        {
            if (s[count] == char_t)
            {
                count += 1;
            }
            if (count >= s.size())
            {
                break;
            }
        }
        return count >= s.size();
    }
};

File: Leetcode/Python/392. Is Subsequence/solution.py
# https://leetcode.com/problems/is-subsequence/

class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        index = 0
        len_s = len(s)
        if len_s == 0:
            return True
        for string in t:
            if string == s[index]:
                index += 1
            if index >= len_s:
                return True
        return False